js/scorm/wrapper.js

```diff
@@ -301,14 +306,31 @@ define ([
 
             this.endTime = new Date();
 
+            var exit_property = "cmi.core.exit";
+            var exit_value = "";
+            var is_complete = this.scorm.data.completionStatus === "completed" || this.scorm.data.completionStatus === "passed";
+
             if (this.isSCORM2004()) {
                 this.scorm.set("cmi.session_time", this.convertToSCORM2004Time(this.endTime.getTime() - this.startTime.getTime()));
-            }
-            else {
+                exit_property = "cmi.exit";
+
+                if (is_complete) {
+                    exit_value = this.exitStateIfComplete === "auto" ? "normal" : this.exitStateIfComplete;
+                } else {
+                    exit_value = this.exitStateIfIncomplete === "auto" ? "suspend" : this.exitStateIfIncomplete;
+                }
+            } else {
                 this.scorm.set("cmi.core.session_time", this.convertToSCORM12Time(this.endTime.getTime() - this.startTime.getTime()));
-                this.scorm.set("cmi.core.exit", "");
+
+                if (is_complete) {
+                    exit_value = this.exitStateIfComplete === "auto" ? "" : this.exitStateIfComplete;
+                } else {
+                    exit_value = this.exitStateIfIncomplete === "auto" ? "" : this.exitStateIfIncomplete;
+                }
             }
 
+            this.scorm.set(exit_property, exit_value);
```

Comment:
it feels a little clumsy to have this here, I think it would actually be clearer to have it set inside the if...else code block above (similar to how `session_time` is handled), then you could drop the `exit_property` variable.